### PR TITLE
Show root formula if roots are irrational

### PR DESCRIPTION
**Pull request by NihadBadalov**
image

### CODE REVIEW

File: QUADROOT.py [hunk 1]
```diff
@@ -13,10 +16,16 @@
...
-	sol1 = (-b - d**0.5) / (2*a)
-	sol2 = (-b + d**0.5) / (2*a)
-	print('\nX = {}\nX = {}'.format(sol1, sol2))
+    # If the discriminant is positive, the solutions are real and different
+    # If we have an irrational number, display the fraction
+    sol1 = (-b - d**0.5) / (2*a)
+    sol2 = (-b + d**0.5) / (2*a)
+    if decimal_point_amount(sol1) > 5:
+        print('\n-  -  -  -  -  -  -')
+        print(f'\nX = ( {-b} + {f"sqrt({d})" if decimal_point_amount(d**0.5) > 5 else {d**0.5}} ) / {2*a}\nX = ( {-b} - {f"sqrt({d})" if decimal_point_amount(d**0.5) > 5 else {d**0.5}} ) / {2*a}')
```
Comment (AbdelRahmanRahal) [inline QUADROOT.py:25]: f-strings do not work on MicroPython v1.9.4, which is the version supported by Casio fx-9750GIII.

File: QUADROOT.py [hunk 2]
```diff
@@ -3,6 +3,9 @@
 print("quadratic form:")
 print("aX^2 + bX + c = 0\n")
 
+def decimal_point_amount(n: float) -> int:
```
Comment (AbdelRahmanRahal) [inline QUADROOT.py:6]: Remove type annotations for backwards compatibility for Casio fx-9750GIII calculators.

**Review by AbdelRahmanRahal — CHANGES_REQUESTED** [review]
This actually used to be how I had it, but I realised I only needed to show the root formulas for my assignments if they were imaginary. Still, it's an nice option. Fix these and I am 100% down.

File: QUADROOT.py (hunk 2 above)
Comment (NihadBadalov) [reply QUADROOT.py]: removed type annotations

File: QUADROOT.py (hunk 1 above)
Comment (NihadBadalov) [reply QUADROOT.py]: f-strings -> format() ✅

**Review by AbdelRahmanRahal — APPROVED** [review]
LGTM! I'll tweak some stuff later to make it look better on Casio displays. Good stuff!

**Comment by NihadBadalov** [comment]
Epic

**Pull request merged**